[ISSUE-46] - Jdbc Realm Definitions.

pedroigor opened the pull request:
Build is failing for this PR. The reason is that I need to figure out how to avoid error due to unavailable capabilities required for our services. The ```JdbcRealmService``` requires the ```DataSource``` API.

I've done some functional testing connecting to a running database to authenticate users. Everything seems to work so far. Specially the capabilities related code.

darranl commented:
Depends on this PR https://github.com/wildfly/wildfly/pull/7682

darranl commented:
Pedro - can you please adjust the commit message so it is in a format GitHub understands: -

https://help.github.com/articles/closing-issues-via-commit-messages/

pedroigor commented:
I've added support to our tests in order to initialize any required capability.

@darranl Good for review and merge.

darranl commented:
Ok that is a little too far 'Fixes #46 JDBC Realm Backed Defintions' would be a more appropriate commit message ;-)

Although it looks like this may need a rebase.

pedroigor commented:
The PR title is already 'JDBC Realm Backed Definitions'. So commits inherit the intend of the PR :) Anyway, I've changed the commit message and rebased. From now on, I'll keep this policy in mind.

Thanks !

darranl commented on src/main/java/org/wildfly/extension/elytron/JdbcRealmDefinition.java:
```diff
@@ -0,0 +1,529 @@
+        }
+    }
+
+    interface PasswordMapperResourceDefinition {
```
I think 'ResourceDefinition' is the wrong term to use here as that has a distinct definition already - maybe 'PasswordMapperObjectDefinition' and also rename the implementations?

pedroigor replied:
Yeah, makes more sense. Changed names accordingly.
